plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java

```diff
@@ -297,8 +297,24 @@ public void testFilterDerivedFromTableProperties()
     @Test
     public void testQueryScanningForTooManyPartitions()
     {
-        assertThatThrownBy(() -> plan("SELECT l.str_col, r.str_col FROM table_int_with_too_many_partitions l JOIN table_unpartitioned r ON l.int_part = r.int_col"))
-                .getCause()
+        String query = "SELECT l.str_col, r.str_col FROM table_int_with_too_many_partitions l JOIN table_unpartitioned r ON l.int_part = r.int_col";
+        // The partitions will be loaded during split creation so planning phase will be successful.
+        assertDistributedPlan(
+                query,
+                noJoinReordering(),
+                output(
+                        exchange(REMOTE, GATHER,
+                                join(INNER, List.of(equiJoinClause("L_INT_PART", "R_INT_COL")),
+                                        exchange(REMOTE, REPARTITION,
+                                                project(
+                                                        filter("true", //dynamic filter
+                                                                tableScan("table_int_with_too_many_partitions", Map.of("L_INT_PART", "int_part", "L_STR_COL", "str_col"))))),
+                                        exchange(LOCAL,
+                                                exchange(REMOTE, REPARTITION,
+                                                        project(
+                                                                tableScan("table_unpartitioned", Map.of("R_STR_COL", "str_col", "R_INT_COL", "int_col")))))))));
+
+        assertThatThrownBy(() -> getQueryRunner().execute(query))
```

Comment:
Why does the query pass only if you disable join reordering?

Reply:
1. this should be a code comment
2. that means the actual user problem has not been solved, right?

Reply:
> that means the actual user problem has not been solved, right?

No. The actual problem that we are trying to solve is - 

Query currently fails when we apply a criteria like `partition_column like '%abc%'` - if the initial number of partitions crosses the threshold - even if only a lesser number of partitions satisfies it.

Reply:
Where is the test showing the successful execution of a query benefiting from the changes here?
i see one in `io.trino.plugin.hive.BaseHiveConnectorTest#testPartitionPerScanLimitWithMultiplePartitionColumns` but it's simple SELECT (good test case to test)
i also see `io.trino.plugin.hive.optimizer.TestHivePlans#testQueryScanningForTooManyPartitions` covering Joins, but it fails with defaults, and requires disabling join reordering to pass. 

The solution should work with defaults setting, or any other setting that are probable to be used by a user.

Reply:
We have fixed for fetching table statistics - now it passes during planning while it fails during execution - due to scanning maximum partitions.